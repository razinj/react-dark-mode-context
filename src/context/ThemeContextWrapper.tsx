import { useState, useEffect, FC, ReactNode } from 'react'
import ThemeContext from './ThemeContext'

const ThemeContextWrapper: FC<{ children: ReactNode }> = ({ children }) => {
  const [theme, setTheme] = useState(localStorage.getItem('theme') || 'light')

  const changeCurrentTheme = (theme: 'light' | 'dark') => {
    setTheme(theme)
    localStorage.setItem('theme', theme)
  }

  useEffect(() => {
    if (theme === 'light') document.body.classList.remove('dark')
    else document.body.classList.add('dark')
  }, [theme])

  return <ThemeContext.Provider value={{ currentTheme: theme, changeCurrentTheme }}>{children}</ThemeContext.Provider>
}

export default ThemeContextWrapper
