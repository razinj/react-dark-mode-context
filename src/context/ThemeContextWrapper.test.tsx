import { render } from '@testing-library/react'
import ThemeContextWrapper from './ThemeContextWrapper'

describe('ThemeContextWrapper', () => {
  let componentContainer: HTMLElement

  beforeEach(() => {
    const { container } = render(
      <ThemeContextWrapper>
        <p>Hello World!</p>
      </ThemeContextWrapper>
    )
    componentContainer = container
  })

  it('should match snapshot', () => {
    expect(componentContainer).toMatchSnapshot()
  })

  it('should render and call switching theme method correctly', () => {
    expect(document.body).not.toHaveClass('dark')
  })
})
