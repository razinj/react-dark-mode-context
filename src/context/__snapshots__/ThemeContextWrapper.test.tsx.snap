// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeContextWrapper should match snapshot 1`] = `
<div>
  <p>
    Hello World!
  </p>
</div>
`;
